lib/src/logger.rs

```diff
@@ -74,10 +142,41 @@ macro_rules! debug_ { ($($args:expr),+) => { log_!(debug: $($args),+); }; }
 #[doc(hidden)] #[macro_export]
 macro_rules! warn_ { ($($args:expr),+) => { log_!(warn: $($args),+); }; }
 
+pub fn format_log(record: &log::Record) -> String {
```

Comment:
Instead of returning a String, this could format directly into the destination to avoid extra allocations, especially in the `Level::Debug` case.

This would mean at minimum taking a `Write` argument and using `write!(writer, ...)` instead of `format!(...)` everywhere.